Gems/GraphCanvas/Code/StaticLib/GraphCanvas/Widgets/NodeSearcher/NodeSearcherDockWidget.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <GraphCanvas/Widgets/NodeSearcher/NodeSearcherDockWidget.h>
+#include <QHeaderView>
+#include <GraphCanvas/Components/VisualBus.h>
+
+namespace GraphCanvas
+{
+    ///////////////////////////
+    // NodeSearcherDockWidget
+    ///////////////////////////
+
+    NodeSearcherDockWidget::NodeSearcherDockWidget(EditorId editorId, QWidget* parent)
+        : AzQtComponents::StyledDockWidget(parent)
+        , m_editorId(editorId)
+    {
+        CreateUI();
+
+        m_model = aznew NodeTableSourceModel();
+        m_proxyModel = aznew NodeTableSortProxyModel(m_model);
+
+        m_nodelistTable->setModel(m_proxyModel);
+         m_nodelistTable->horizontalHeader()->setSectionResizeMode(NodeTableSourceModel::CD_Name, QHeaderView::ResizeMode::Stretch);
+        m_nodelistTable->verticalHeader()->setVisible(false);
+
+        m_filterTimer.setInterval(250);
+        m_filterTimer.setSingleShot(true);
+        m_filterTimer.stop();
+
+        connect(&m_filterTimer, &QTimer::timeout, this, &NodeSearcherDockWidget::UpdateFilter);
+        connect(m_quickFilter, &QLineEdit::textChanged, this, &NodeSearcherDockWidget::OnQuickFilterChanged);
+        connect(m_nodelistTable->selectionModel(), &QItemSelectionModel::selectionChanged, this, &NodeSearcherDockWidget::SelectionChanged);
+        connect(m_nodelistTable, &QTableView::doubleClicked, this, &NodeSearcherDockWidget::OnDoubleClicked);
+        AssetEditorNotificationBus::Handler::BusConnect(editorId);
+
+        OnActiveGraphChanged(AZ::EntityId());
+    }
+    NodeSearcherDockWidget::~NodeSearcherDockWidget()
+    {
+        AssetEditorNotificationBus::Handler::BusDisconnect(m_editorId);
+    }
```

Comment:
Make sure to disconnect from the scene notification bus and any other buses in the destructor. Since you are disconnecting from everything you don't need to pass in the id.